int32 flag
float64 dist_x
float64 dist_y
